### PR TITLE
feat/AB#106560_EIS-improve-activity-log-navigation

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

feat: make sure that the application id is fetch if exists on application page enter 
feat: remove not needed method from logger service

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/106560)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

Please include screenshots of this change. If this issue is only back-end related, and does not involve any visual change of the platform, you can skip this part.

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by unai-reliefapp** [comment]
@AntoineRelief first view on the application thingie, let me know if the approach could be fine

File: libs/shared/src/lib/services/application/application.service.ts [hunk 1]
```diff
@@ -291,6 +301,7 @@ export class ApplicationService {
         error: () => {
           this.hasErrors = true;
           this.application.next(null);
+          this.loadingApplication.next(false);
```
Comment (AntoineRelief) [inline libs/shared/src/lib/services/application/application.service.ts:304]: @unai-reliefapp 
perhaps a finally should do it?

Comment (unai-reliefapp) [reply libs/shared/src/lib/services/application/application.service.ts:304]: complete function is not triggered if an error is launched, Ive been trying but no luck :(

Comment (AntoineRelief) [reply libs/shared/src/lib/services/application/application.service.ts:304]: okay then it's good like that

**Comment by AntoineRelief** [comment]
@unai-reliefapp 

it looks quite better like that 👍 

there may be a edge case, and I don't know how often it can happen, but we need to be sure that:
- if you exit an application ( back-office ), then you don't set applicationId in metadata ( so the app service doesn't contain app anymore )
- especially if, from the back-office, you enter an app and quickly go back ( without letting time for the system to load the application )
- if you quickly switch between applications, make sure that the latest application call is taken into account

in fact, you may need to update the apollo call inside the loadApplication method to use a takeUntil, based on the loadingApplication?
so if you emit a new value while loading an app, you know you need to cancel the first apollo query, and only focus on the second one

hope that makes sense!

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp
> 
> it looks quite better like that 👍
> 
> there may be a edge case, and I don't know how often it can happen, but we need to be sure that:
> 
> * if you exit an application ( back-office ), then you don't set applicationId in metadata ( so the app service doesn't contain app anymore )
> * especially if, from the back-office, you enter an app and quickly go back ( without letting time for the system to load the application )
> * if you quickly switch between applications, make sure that the latest application call is taken into account
> 
> in fact, you may need to update the apollo call inside the loadApplication method to use a takeUntil, based on the loadingApplication? so if you emit a new value while loading an app, you know you need to cancel the first apollo query, and only focus on the second one
> 
> hope that makes sense!

I think that the takeUntil with the loadingApplication emit would take care of those cases, I'll update the PR

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp
> 
> it looks quite better like that 👍
> 
> there may be a edge case, and I don't know how often it can happen, but we need to be sure that:
> 
> * if you exit an application ( back-office ), then you don't set applicationId in metadata ( so the app service doesn't contain app anymore )
> * especially if, from the back-office, you enter an app and quickly go back ( without letting time for the system to load the application )
> * if you quickly switch between applications, make sure that the latest application call is taken into account
> 
> in fact, you may need to update the apollo call inside the loadApplication method to use a takeUntil, based on the loadingApplication? so if you emit a new value while loading an app, you know you need to cancel the first apollo query, and only focus on the second one
> 
> hope that makes sense!

Actually the takeuntil is not needed, the leaveApplication method unsubscribes automatically from any ongoing application fetch
https://github.com/ReliefApplications/ems-frontend/blob/AB%23105244_activity_logs/libs/shared/src/lib/services/application/application.service.ts#L348
which is called on each loadapplication
https://github.com/ReliefApplications/ems-frontend/blob/AB%23105244_activity_logs/libs/shared/src/lib/services/application/application.service.ts#L246

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
wonderful!
I guess we had this issue before, so that's how we solved it
looks good then, please finish the PR and assign to me 👍

**Comment by unai-reliefapp** [comment]
@AntoineRelief Explanation in the logger service as a comment for further references
also:
https://github.com/user-attachments/assets/b6a30500-1eeb-41f9-bb1e-0cbcffc6406e

**Pull request merged**